spring-session/src/main/java/org/springframework/session/data/mongo/MongoSessionRepository.java

```diff
@@ -0,0 +1,160 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.mongo;
+
+import com.mongodb.DBObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.session.MapSession;
+import org.springframework.session.events.SessionCreatedEvent;
+import org.springframework.session.events.SessionDeletedEvent;
+import org.springframework.session.events.SessionExpiredEvent;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Session repository implementation which stores sessions in Mongo.
+ * Uses {@link MongoSessionSerializer} to transform session objects from/to
+ * native Mongo representation ({@code DBObject}).
+ *
+ * Repository is also responsible for removing expired sessions from database.
+ * Cleanup is done every minute.
+ *
+ * @author Jakub Kubrynski
+ */
+public class MongoSessionRepository implements FindByIndexNameSessionRepository<ExpiringSession> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(MongoSessionRepository.class);
+
+	private static final String ID_FIELD_NAME = "_id";
+
+	private final MongoOperations mongoOperations;
+	private final ApplicationEventPublisher eventPublisher;
+	private final MongoSessionSerializer sessionSerializer;
+	private final Integer maxInactiveIntervalInSeconds;
+	private final String collectionName;
+
+	public MongoSessionRepository(MongoOperations mongoOperations, ApplicationEventPublisher eventPublisher,
+	                              MongoSessionSerializer mongoSessionSerializer, Integer maxInactiveIntervalInSeconds, String collectionName) {
+		this.mongoOperations = mongoOperations;
+		this.eventPublisher = eventPublisher;
+		this.sessionSerializer = mongoSessionSerializer;
+		this.maxInactiveIntervalInSeconds = maxInactiveIntervalInSeconds;
+		this.collectionName = collectionName;
+	}
+
+	@Override
+	public ExpiringSession createSession() {
+		MapSession session = new MapSession(UUID.randomUUID().toString().replaceAll("-", ""));
```

Comment:
Just making the id shorter. `-` sign is redundant here in my opinion. Of course it will also work with minus char so if you prefer this version I can remove the replacement